server/src/daemon/prompter.rs

```diff
@@ -0,0 +1,210 @@
+    continue_label: Option<String>,
+    #[zvariant(rename = "cancel-label")]
+    cancel_label: Option<String>,
+    #[zvariant(rename = "choice-chosen")]
+    choice_chosen: Option<bool>,
+    #[zvariant(rename = "password-new")]
+    password_new: Option<bool>,
+    #[zvariant(rename = "password-strength")]
+    password_strength: Option<u32>,
+}
+
+// org.gnome.keyring.internal.Prompter.Callback
+
+#[derive(Clone, Debug)]
+pub struct PrompterCallback {
+    properties: Arc<Mutex<Properties>>,
+    path: OwnedObjectPath,
+}
+
+#[interface(name = "org.gnome.keyring.internal.Prompter.Callback")]
+impl PrompterCallback {
+    pub async fn prompt_ready(
+        &self,
+        reply: &str,            // the purpose of this?
+        properties: Properties, // this should probably be &mut Properties
+        exchange: &str,         // this should probably be &mut String
+        #[zbus(connection)] connection: &zbus::Connection,
+        #[zbus(header)] header: Header<'_>,
+    ) -> fdo::Result<()> {
+        println!("prompter: {}", header.path().unwrap());
+
+        let mut properties = properties.clone();
+
+        if header.path().unwrap().as_str().to_string().contains("/u") {
+            // setting properties related to Secret.Service.Unlock
+            properties.continue_label = Some(String::from("Unlock"));
+            properties.warning = Some(String::new());
+            properties.choice_chosen = Some(true);
+            properties.description = Some(String::from(
+                "An application wants access to the keyring \"login\", but it is locked.",
+            ));
+            properties.title = Some(String::from("Unlock Keyring"));
+            properties.message = Some(String::from("Authentication required"));
+            properties.choice_label = Some(String::from(
+                "Automatically unlock this keyring whenever I'm logged in",
+            ));
+            properties.caller_window = Some(String::new());
+            properties.cancel_label = Some(String::from("Cancel"));
+        } else {
+            // setting properties related to Secret.Service.CreateCollection
+            properties.continue_label = Some(String::from("Continue"));
+            properties.warning = Some(String::new());
+            properties.choice_chosen = Some(false);
+            properties.description = Some(String::from(
+                "An application wants to create a new keyring. Choose the password you want to use for it.",
+            ));
+            properties.title = Some(String::new());
+            properties.message = Some(String::from("Choose password for new keyring"));
+            properties.choice_label = Some(String::new());
+            properties.caller_window = Some(String::new());
+            properties.cancel_label = Some(String::from("Cancel"));
+            properties.password_new = Some(true);
+        }
+
+        *self.properties.lock().unwrap() = properties.clone();
```

Comment:
Why you need to store the properties here?

Reply:
it was there just temporarily, now removed.